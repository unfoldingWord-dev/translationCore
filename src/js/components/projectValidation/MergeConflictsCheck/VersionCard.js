import React, { Component } from 'react';
import { RadioButton } from 'material-ui/RadioButton';
import PropTypes from 'prop-types';

class VersionCard extends Component {
  getTextObjectSection(verseData) {
    return Object.keys(verseData).map((verseNumber) => (
      <div style={{ fontSize: 14 }} key={verseNumber}>
        <b>{verseNumber}</b>: {verseData[verseNumber]}
      </div>
    ));
  }

  render() {
    let {
      checked, index, mergeConflictIndex, textData, translate,
    } = this.props;
    return (
      <div style={{ borderBottom: '1px solid black' }}>
        <div style={{ padding: 15 }}>
          <RadioButton
            checked={checked}
            label={translate('version', { 'version': Number(index) + 1 })}
            onCheck={() => this.props.onCheck(mergeConflictIndex, index, true)}
          />
          {this.getTextObjectSection(textData)}
        </div>
      </div>
    );
  }
}

VersionCard.propTypes = {
  index: PropTypes.string.isRequired,
  mergeConflictIndex: PropTypes.string.isRequired,
  textData: PropTypes.object.isRequired,
  onCheck: PropTypes.func.isRequired,
  checked: PropTypes.any,
  translate: PropTypes.func.isRequired,
};

export default VersionCard;