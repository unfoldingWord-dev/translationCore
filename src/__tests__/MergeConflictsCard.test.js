/* eslint-env jest */
import React from 'react';
import PropTypes from 'prop-types';
import renderer from 'react-test-renderer';
import {
  shallow, mount, configure,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import RightArrow from 'material-ui/svg-icons/hardware/keyboard-arrow-right';
import DownArrow from 'material-ui/svg-icons/hardware/keyboard-arrow-down';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import VersionCard from '../js/components/projectValidation/MergeConflictsCheck/VersionCard';
import MergeConflictsCard from '../js/components/projectValidation/MergeConflictsCheck/MergeConflictsCard';
import MergeConflicts from '../js/components/projectValidation/MergeConflictsCheck/';

describe('MergeConflictsComponents.index', () => {
  const wrapperOptions = {
    context: { muiTheme: getMuiTheme() },
    childContextTypes: { muiTheme: PropTypes.object },
  };

  beforeAll(() => {
    configure({ adapter: new Adapter() });
  });

  const initialState = {
    updateVersionSelection: jest.fn(),
    translate: key => key,
    reducers: {
      mergeConflictReducer: {
        conflicts: [
          [{
            chapter: '1',
            text: {
              5: 'This verse has been changed by another user',
              6: 'Some other verse',
            },
            verses: '5-6',
            checked: false,
          },
          {
            chapter: '1',
            text: {
              5: 'Some verse',
              6: 'Some other verse',
            },
            verses: '5-6',
            checked: true,
          }],
        ],
      },
    },
  };

  it('should render the default component and match the snapshot', () => {
    const wrapper = renderer.create(
      <MuiThemeProvider>
        <MergeConflicts {...initialState} />
      </MuiThemeProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the expeted props given the state', () => {
    const wrapper = mount(
      <MuiThemeProvider>
        <MergeConflicts {...initialState} />
      </MuiThemeProvider>
      , wrapperOptions);

    expect(wrapper.find(MergeConflictsCard).props()).toEqual({
      translate: expect.any(Function),
      chapter: '1',
      verses: '5-6',
      mergeConflictIndex: '0',
      versions:
        [{
          index: '0', textData: initialState.reducers.mergeConflictReducer.conflicts[0][0].text, checked: false,
        },
        {
          index: '1', textData: initialState.reducers.mergeConflictReducer.conflicts[0][1].text, checked: true,
        }],
      open: false,
      onCheck: expect.any(Function),
      openCard: expect.any(Function),
    });
  });

  it('should open merge conflict card', () => {
    const wrapper = shallow(
      <MergeConflicts {...initialState} />
      , wrapperOptions);
    expect(wrapper.find(MergeConflictsCard).prop('open')).toBe(false);
    wrapper.setState({ conflictCards: [{ open: true }] });
    expect(wrapper.find(MergeConflictsCard).prop('open')).toBe(true);
  });
});



describe('MergeConflictsComponents.MergeConflictsCard', () => {
  const wrapperOptions = {
    context: { muiTheme: getMuiTheme() },
    childContextTypes: { muiTheme: PropTypes.object },
  };

  beforeAll(() => {
    configure({ adapter: new Adapter() });
  });

  const initialState = {
    translate: key => key,
    chapter: '1',
    verses: '5-6',
    mergeConflictIndex: '0',
    versions:
      [{
        index: '0',
        textData: {
          5: 'This verse has been changed by another user',
          6: 'Some other verse',
        },
        checked: false,
      },
      {
        index: '1',
        textData: {
          5: 'Some verse',
          6: 'Some other verse',
        },
        checked: true,
      }],
    open: true,
    onCheck: () => {},
    openCard: () => {},
  };

  it('should show merge conflicts versions if open', () => {
    const wrapper = shallow(
      <MergeConflictsCard {...initialState} />
      , wrapperOptions);
    expect(wrapper.find(VersionCard).at(0)).toMatchSnapshot();
    expect(wrapper.find(VersionCard).at(1)).toMatchSnapshot();
    expect(wrapper.find(RightArrow)).toHaveLength(1);
    expect(wrapper.find(DownArrow)).toHaveLength(0);
    wrapper.setProps({ open: false });
    expect(wrapper.find(RightArrow)).toHaveLength(0);
    expect(wrapper.find(DownArrow)).toHaveLength(1);
    expect(wrapper.find(VersionCard)).toHaveLength(0);
  });
});
