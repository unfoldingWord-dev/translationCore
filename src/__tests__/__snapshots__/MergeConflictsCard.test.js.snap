// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MergeConflictsComponents.MergeConflictsCard should show merge conflicts versions if open 1`] = `ShallowWrapper {}`;

exports[`MergeConflictsComponents.MergeConflictsCard should show merge conflicts versions if open 2`] = `ShallowWrapper {}`;

exports[`MergeConflictsComponents.index should render the default component and match the snapshot 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
      "height": "85%",
      "marginTop": "10px",
    }
  }
>
  <div
    style={
      Object {
        "height": "100%",
        "minWidth": "400px",
        "padding": "0px 20px 0 65px",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "height": "100%",
        }
      }
    >
      instructions
      <div
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "#ffffff",
            "borderRadius": 2,
            "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "Roboto, sans-serif",
            "height": "100%",
            "lineHeight": "2em",
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "zIndex": 1,
          }
        }
      >
        <div
          style={
            Object {
              "paddingBottom": 8,
            }
          }
        >
          <div
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "fontSize": 14,
                "height": "100%",
                "muiPrepared": true,
                "padding": 16,
              }
            }
          >
            <div>
              project_validation.merge_conflicts_found
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "height": "100%",
        "padding": "0px 50px 22px 20px",
        "width": "100%",
      }
    }
  >
    <div
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      project_validation.merge_conflicts
      <div
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "#ffffff",
            "borderRadius": 2,
            "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "Roboto, sans-serif",
            "height": "100%",
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%",
            "zIndex": 1,
          }
        }
      >
        <div
          style={
            Object {
              "height": "100%",
              "overflowY": "auto",
              "paddingBottom": 0,
            }
          }
        >
          <div
            style={
              Object {
                "borderBottom": "1px solid black",
                "paddingBottom": 20,
              }
            }
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                }
              }
            >
              <div
                style={
                  Object {
                    "padding": "15px 15px 0px 15px",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "fontWeight": "bold",
                      "paddingBottom": 5,
                    }
                  }
                >
                  project_validation.merge_conflict
                </div>
                <div>
                  project_validation.conflict_reference
                </div>
              </div>
              <div
                onClick={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "border": "2px solid black",
                    "borderRadius": "50%",
                    "cursor": "pointer",
                    "display": "flex",
                    "height": 40,
                    "justifyContent": "center",
                    "margin": "15px 15px 0px auto",
                    "width": 40,
                  }
                }
              >
                <svg
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "color": "rgba(0, 0, 0, 0.87)",
                      "display": "inline-block",
                      "fill": "currentColor",
                      "height": 60,
                      "muiPrepared": true,
                      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                      "userSelect": "none",
                      "width": 60,
                    }
                  }
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7.41 7.84L12 12.42l4.59-4.58L18 9.25l-6 6-6-6z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
